File: find-peak-element.py
class Solution:
    # @param nums, an integer[]
    # @return an integer
    def findPeakElement(self, nums):
        if len(nums) <= 1:
            return 0
        for i in range(0, len(nums)):
            if i == 0:
                if nums[i] > nums[i+1]:
                    return i
            elif i == len(nums) - 1:
                if nums[i] > nums[i-1]:
                    return i
            else:
                if nums[i] > nums[i-1] and nums[i] > nums[i+1]:
                    return i
                

File: src/main/cpp/find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        if (nums.size() <= 1) return 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (i == 0 && nums[i] > nums[i+1]) return 0;
            else if (i == nums.size() - 1 && nums[i] > nums[i-1]) return nums.size() - 1;
            else if (i > 0 && i < nums.size() - 1) {
                if (nums[i] > nums[i-1] && nums[i] > nums[i+1]) return i;
            }
        }
    }
};